src/DropdownReference/components/DropdownReferenceContainer.ts

```diff
@@ -0,0 +1,252 @@
+    }
+
+    private isReadOnly = (): boolean => {
+        const { editable, mxObject, readOnly } = this.props;
+
+        return editable !== "default" || !mxObject || readOnly;
+    }
+
+    private resetSubscriptions(mxObject?: mendix.lib.MxObject) {
+        this.subscriptionHandles.forEach(window.mx.data.unsubscribe);
+        this.subscriptionHandles = [];
+
+        if (mxObject) {
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                attr: this.association,
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                attr: this.props.attribute,
+                callback: this.handleSubscriptions,
+                guid: mxObject.get(this.association) as string
+            }));
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                entity: this.props.entityPath.split("/")[1],
+                callback: () => {
+                    this.setState({ selectedObject: null });
+                    this.getSelectedValue(this.props);
+                    this.retrieveOptions(this.props);
+                }
+            }));
+        }
+    }
+
+    private handleSubscriptions = () => {
+        this.getSelectedValue(this.props);
+    }
+
+    private onChange = (selection: ReferenceOption) => {
+        if (this.props.mxObject) {
+
+            if (!selection) {
+                const references = this.props.mxObject.getReferences(this.association);
+                this.props.mxObject.removeReferences(this.association, references);
+            } else {
+                this.props.mxObject.set(this.association, selection.value);
+                if (!this.state.selectedObject || (this.state.selectedObject.value !== selection.value)) {
+                    this.executeOnChangeEvent();
+                }
+            }
+        }
+
+        this.setState({ selectedObject: selection ? selection : null });
+    }
+
+    private executeOnChangeEvent = () => {
+        const { mxform, mxObject, onChangeEvent, onChangeMicroflow, onChangeNanoflow } = this.props;
+        const context = new mendix.lib.MxContext();
+
+        context.setContext(mxObject.getEntity(), mxObject.getGuid());
+        if (onChangeEvent === "callMicroflow" && onChangeMicroflow) {
+            window.mx.ui.action(onChangeMicroflow, {
+                error: error => window.mx.ui.error(`Error while executing onchange microflow ${onChangeMicroflow}: ${error.message}`), // tslint:disable-line max-line-length
+                origin: mxform,
+                params: {
+                    applyto: "selection",
+                    guids: [ mxObject.getGuid() ]
+                }
+            });
```

Comment:
prefer order of importance, as documented
``` ts
window.mx.ui.action(onChangeMicroflow, {
	params: {
		applyto: "selection",
		guids: [ mxObject.getGuid() ]
	},
	origin: mxform
	error: error => window.mx.ui.error(`Error while executing onchange microflow ${onChangeMicroflow}: ${error.message}`)
});
```

Reply:
Done